### ISSUE TITLE
feat: kubie integration or ability to have isolated context

### ISSUE THREAD

**Issue by jemag** [Ramilito/kubectl.nvim#528]
**Is your feature request related to a problem? Please describe.**
Currently changing the context within `kubectl.nvim` will change the overall context for kubectl.

**Describe the solution you'd like**
I would like to be able to isolate the context change to only happen within `kubectl.nvim`. This could be done through an integration with [kubie](https://github.com/sbstp/kubie) or with its own setup.

I often keep my `kubectl` cli on a specific sandbox cluster, while using `kubectl.nvim` to explore and debug other clusters. However, currently my "default" cluster will continuously change due to context changes within `kubectl.nvim`. 

**Describe alternatives you've considered**
Load kubie in a specific cluster before opening kubectl.nvim. However this has the problem that only this specific cluster will be available in kubectl.nvim context, so cannot easily change afterwards.

**Additional context**

**Comment by Ramilito** [comment]
Yeah I would love to have per session as well, there sort of already is since we use the kubectl proxy that persists the connection even if you would change the context (you can test it out using multiple nvim instances).
The simplest solution would be to adjust the change context command to just create a proxy using the --context flag instead of assuming the current context.

**Comment by Ramilito** [comment]
@mosheavni, @jemag If we make that change (to not actually change the context but instead always use that proxy), we should probably make it possible to persist that change some how, do you have any ideas how to make that UX wise?

**Comment by Ramilito** [comment]
> most plugins seem to use ~/.local/state/nvim/{plugin-name} to store any state that should persist. Common pattern is to either have it stored in json or lua file. I think both approaches are valid.

> I assume that the persisting would mean once a user change context within one kubectl.nvim instance, further newly created kubectl.nvim instances would use that same context?

I agree here, we already save the current session, history and other things in that place so we could add last_context as well to keep track of it and avoid actually changing the default context and instead using the `kubectl proxy --context` to specify what context we are connecting to. 

There is no real problem implementation wise but rather UX wise. Do we want to offer a way to change the default context?
How confusing would it be if you open a `kubectl.nvim` session to a different context than your `kubectl` has? Or, close `kubectl.nvim` to find that your shell is using a different one?

**Comment by jemag** [comment]
I dont find it particularly confusing since I am used to that workflow, but that could indeed be the case for existing users.

Perhaps this behavior could simply be configured through an option? (i.e.: retain current behavior or have isolated context)

**Comment by Ramilito** [comment]
Maybe a combination of configuration and a toggle, the configuration could set the default value of the toggle?

#Configuration would be  
```
persist_context_change = true
```

And then in the change context view we would have a way to change that for just that change, it's mostly that I feel that just having a configuration and force people to restart nvim might be annoying

**Comment by jemag** [comment]
sounds good to me

**Comment by Ramilito** [comment]
Just a FYI, I've written a tool that is similar to `kubie` called [kubesess](https://github.com/Ramilito/kubesess) that works with this plugin as long as you don't change the context using the plugin, you can have different contexts at the same time

**Comment by Ramilito** [comment]
This should work now!
